pagai/services/database_explorer.py

```diff
@@ -96,45 +96,68 @@ def check_connection_exists(self):
     def get_sql_alchemy_table(self, table: str):
         return Table(table.strip(), self._metadata, schema=self.owner, autoload=True)
 
-    def get_sql_alchemy_column(self, column: str, table: str):
+    def get_sql_alchemy_column(self, column: str, sqlalchemy_table: Table):
         """
         Return column names of a table
         """
-        table = self.get_sql_alchemy_table(table)
         try:
-            return table.c[column]
+            return sqlalchemy_table.c[column]
         except KeyError:
             # If column is not in table.c it may be because the column names
             # are case insensitive. If so, the schema can be in upper case
             # (what oracle considers as case insensitive) but the keys
             # in table.c are in lower case (what sqlalchemy considers
             # as case insensitive).
-            return table.c[column.lower()]
+            return sqlalchemy_table.c[column.lower()]
 
-    def get_table_rows(self, table_name: str, limit=100, filters=[]):
+    def get_table_rows(self, session, table_name: str, limit=100, filters=[]):
         """
         Return content of a table with a limit
         """
+        columns_names = self.db_schema[table_name]
+
         table = self.get_sql_alchemy_table(table_name)
-        select = table.select()
+        columns = [self.get_sql_alchemy_column(col, table) for col in columns_names]
+        select = session.query(*columns)
 
         # Add filtering if any
         for filter_ in filters:
-            col = self.get_sql_alchemy_column(
-                filter_["sqlColumn"]["column"], filter_["sqlColumn"]["table"]
-            )
+            table = self.get_sql_alchemy_table(filter_["sqlColumn"]["table"])
+            col = self.get_sql_alchemy_column(filter_["sqlColumn"]["column"], table)
             filter_clause = SQL_RELATIONS_TO_METHOD[filter_["relation"]](col, filter_["value"])
-            select = select.where(filter_clause)
-
-        columns_names = self.db_schema[table_name]
+            select = select.filter(filter_clause)
+
+            # Apply joins
+            # TODO use fhir-river's analyzer?
+            # TODO I don't think it properly handles several filters
+            # with same join
```

Comment:
is this hard to unit-test ? I doesn't seem easy but maybe you can think of something ?

Reply:
I don't think it would be an issue if the same table is joined twice even if it's on a different column (the joined table would be aliased or something, no ?)

Reply:
Yeah, I don't exactly know why I wrote that 🤷‍♂️

Reply:
keep the comment if you're still a little bit worried, remove it otherwise